docs/resources/ui-building-blocks/components/user-defined-components/component-lifecycle.md

```diff
@@ -0,0 +1,256 @@
+caching data pertinent to the component's functionality.
+
+For example,
+
+- In a custom drop-down menu component, you might use a component state variable to keep track of
+  which item is currently selected.
+- In a toggle switch component, you could use a component state variable to store the on/off
+  state based on user interaction.
+  This approach ensures that the state of the component is maintained as it interacts with the user
+  or other parts of the application.
+
+When the value of a component state variable is changed, the component can be re-rendered with the
+updated values, displaying the latest state of the component with these updates.
+
+### Creating a Component State
+
+To create a new **Component State variable** in your component, follow the steps:
+
+
+<div style={{
+    position: 'relative',
+    paddingBottom: 'calc(56.67989417989418% + 41px)', // Keeps the aspect ratio and additional padding
+    height: 0,
+    width: '100%'
+}}>
+    <iframe 
+        src="https://demo.arcade.software/nEmCDqupF7YHUTi4hKvW?embed&show_copy_link=true"
+        title="Create Component State"
+        style={{
+            position: 'absolute',
+            top: 0,
+            left: 0,
+            width: '100%',
+            height: '100%',
+            colorScheme: 'light'
+        }}
+        frameborder="0"
+        loading="lazy"
+        webkitAllowFullScreen
+        mozAllowFullScreen
+        allowFullScreen
+        allow="clipboard-write">
+    </iframe>
+</div>
+
+While creating a Component State, the following properties are included:
+
+- **Is List:** This property determines whether the variable can hold multiple values of the same
+  data type (like a list or array) or just a single value.
+
+- **Initial Field Value:** This property sets the default value for the variable when it is first
+  created. It's like setting the starting point or the value that the variable begins with before
+  anything else happens.
+
+- **Nullable:** This property determines whether the variable can have a null value. When "*
+  *Nullable**" is set to true, it means the variable can be empty or have a null value. This is
+  useful when dealing with optional data or scenarios where the absence of a value is valid.
+
+Now, let's apply these concepts to the `isFavourite` variable in the context of the above example:
+
+- For the `isFavourite` variable, it is a single value (boolean), so **Is List** would be set to
+  false.
+
+- The **Initial Field Value** would also be set to **false**, indicating that the item is not
+  favorited by default.
+
+- **Nullable** property will be set to false, as the variable should always have a boolean value
+  (true or false) and never be null.
+
+:::note
+You can set the **Data Type** of your Component State variable to any primitive data types such as *
+*String,
+Integer, Boolean, Double** or to any other complex built-in data types such as **Enum, Custom Data
```

Comment:
**String, Integer, Boolean, Double** shows as it is on a page.

Reply:
Yeah some issue with formatting pages, it just brings them to next line and then the ** ** is not applied properly.